src/Swarm/Language/Typecheck.hs

```diff
@@ -164,7 +165,7 @@ noSkolems l (Forall xs upty) = do
           (\case TyVarF v -> S.singleton v; f -> fold f)
           upty'
       ftyvs = tyvs `S.difference` S.fromList xs
-  forM_ (S.lookupMin ftyvs) $ throwError . EscapedSkolem l
+  forM_ (S.lookupMin ftyvs) $ throwTypeErr l . EscapedSkolem
```

Comment:
Does this get to throw more than once?

I would expect this to boil down to “find at most one unbound variable”.

Reply:
Yes, `S.lookupMin :: Set a -> Maybe a` returns the smallest element of a set, or `Nothing` if the set is empty.  We don't actually care about getting the *smallest* type variable, but `lookupMin` is a convenient way to say "just get me one element if any exist".  The `forM_` is actually over the `Maybe` so it represents doing the `throwTypeErr` either zero or one time, depending on whether `lookupMin` returns `Nothing` or `Just`.

I will add a comment to this effect in the code.